forge/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -0,0 +1,230 @@
+                    mocks.iter().find(|(mock, _)| *mock == &call.input[..mock.len()])
+                {
+                    return (Return::Return, Gas::new(0), mock_retdata.clone())
+                }
+            }
+
+            (Return::Continue, Gas::new(0), Bytes::new())
+        }
+    }
+
+    fn initialize_interp(
+        &mut self,
+        interpreter: &mut Interpreter,
+        data: &mut EVMData<'_, DB>,
+        _: bool,
+    ) -> Return {
+        // Apply our prank
+        if let Some(prank) = &self.prank {
+            if data.subroutine.depth() >= prank.depth &&
+                interpreter.contract.caller == prank.prank_caller
+            {
+                // At the target depth we set `msg.sender`
+                if data.subroutine.depth() == prank.depth {
+                    interpreter.contract.caller = prank.new_caller;
+                }
+
+                // At the target depth, or deeper, we set `tx.origin`
+                if let Some(new_origin) = prank.new_origin {
+                    data.env.tx.caller = new_origin;
+                }
+            }
+        }
+
+        Return::Continue
+    }
+
+    fn step(&mut self, interpreter: &mut Interpreter, _: &mut EVMData<'_, DB>, _: bool) -> Return {
+        if let Some(storage_accesses) = &mut self.accesses {
+            match interpreter.contract.code[interpreter.program_counter()] {
+                opcode::SLOAD => {
+                    let key = try_or_continue!(interpreter.stack().peek(0));
+                    storage_accesses
+                        .reads
+                        .entry(interpreter.contract().address)
+                        .or_insert_with(Vec::new)
+                        .push(key);
+                }
+                opcode::SSTORE => {
+                    let key = try_or_continue!(interpreter.stack().peek(0));
+                    storage_accesses
+                        .writes
+                        .entry(interpreter.contract().address)
+                        .or_insert_with(Vec::new)
+                        .push(key);
+                }
+                _ => (),
+            }
+        }
+
+        Return::Continue
+    }
+
+    fn call_end(
+        &mut self,
+        data: &mut EVMData<'_, DB>,
+        call: &CallInputs,
+        remaining_gas: Gas,
+        status: Return,
+        retdata: Bytes,
+        _: bool,
+    ) -> (Return, Gas, Bytes) {
+        if call.contract == *CHEATCODE_ADDRESS {
+            return (status, remaining_gas, retdata)
+        }
+
+        // Clean up pranks
+        if let Some(prank) = &self.prank {
+            data.env.tx.caller = prank.prank_origin;
+            if prank.single_call {
+                std::mem::take(&mut self.prank);
+            }
+        }
+
+        // Handle expected reverts
+        if let Some(expected_revert) = &self.expected_revert {
+            if data.subroutine.depth() <= expected_revert.depth {
```

Comment:
This depth check is needed to replicate a specific behavior we have on master now, see this test: https://github.com/onbjerg/foundry-test/blob/7909fe8db74ca13fd86cc501fbbb53baaaec5db8/src/cheats/ExpectRevert.t.sol#L79-L84

The reason this was handled in Sputnik without recording the `depth` is because we wrapped the entirety of `call_inner`